updater.go

```diff
@@ -65,16 +65,23 @@ func (u *Updater) Do() error {
 		return fmt.Errorf("unable to stat old binary:%w", err)
 	}
 	mode := info.Mode()
+	//nolint:gosec
 	lf, err := os.OpenFile(location, os.O_WRONLY, mode)
 	if err != nil {
 		if os.IsPermission(err) {
 			return fmt.Errorf("unable to write to:%s need root access:%w", location, err)
 		}
 	}
-	lf.Close()
+	err = lf.Close()
+	if err != nil {
+		return fmt.Errorf("unable to close file:%w", err)
+	}
 
 	oldlocation := location + ".update"
-	defer os.Remove(oldlocation)
+	err = os.Remove(oldlocation)
+	if err != nil {
+		return fmt.Errorf("unable to remove old binary:%w", err)
+	}
```

Comment:
```suggestion
	defer func() {
	err = os.Remove(oldlocation)
	if err != nil {
		return fmt.Errorf("unable to remove old binary:%w", err)
	}
	}()
```

Otherwise you would remove the oldLocation immediatly and this would break the whole logic

Reply:
some for other places